keyboards/nullbitsco/nibble/keymaps/snailmap/keymap.c

```diff
@@ -0,0 +1,552 @@
+            write_pixel(keymap_template[0] + 10 + i, keymap_template[1] + 17, false);
+            // Draw vertical line for ISO enter
+            write_pixel(keymap_template[0] + 79, keymap_template[1] + 8 + i, true);
+        }
+    }
+    // Update locations of shift and grave key highlights
+    keymap_coords[3][1][2]  = 3;
+    keymap_coords[1][14][0] = 70;
+    keymap_coords[1][14][1] = 10;
+    keymap_coords[1][14][2] = 1;
+}
+
+// Toggles pixels surrounding key
+static void render_keymap(uint8_t key_row, uint8_t key_col, bool onoff) {
+    int length = keymap_coords[key_row][key_col][2] + 4;
+    int left   = keymap_coords[key_row][key_col][0] + keymap_template[0];
+    int top    = keymap_coords[key_row][key_col][1] + keymap_template[1];
+    int right  = left + length - 1;
+    int bottom = top + 4;
+    
+    // Special case 1 - Draw enter key on ISO layout, return
+    if ((ansi_layout == false) && (key_row == 2) && (key_col == 14)) {
+        for (int i = 0; i < 10; i++) {
+            write_pixel(keymap_template[0] + 81, keymap_template[1] + 5 + i, onoff);
+            if (i < 5) {
+                write_pixel(keymap_template[0] + 74, keymap_template[1] + 5 + i, onoff);
+            }
+            if (i < 6) {
+                write_pixel(keymap_template[0] + 75, keymap_template[1] + 9 + i, onoff);
+            }
+            if (i < 7) {
+                write_pixel(keymap_template[0] + 75 + i, keymap_template[1] + 5, onoff);
+                write_pixel(keymap_template[0] + 75 + i, keymap_template[1] + 14, onoff);
+            }
+        }
+        return;
+    }
+    // Draw top and bottom walls (horizontal for <length>px)
+    for (int x = 0; x < length; x++) {
+        write_pixel(left + x, top, onoff);
+        write_pixel(left + x, bottom, onoff);
+    }
+    // Draw left and right walls (vertical for 5px)
+    for (int y = 0; y < 5; y++) {
+        write_pixel(left,  top + y, onoff);
+        write_pixel(right, top + y, onoff);
+    }
+    // Special case 2 - Draw right spacebar on split-space layout
+    if ((split_space == true) && (key_row == 4) && (key_col == 6)) {
+        int start = keymap_template[0] + 46;
+        int stop  = keymap_template[0] + 55;
+        for (int x = start; x < stop; x++) {
+            write_pixel(x, top, onoff);
+            write_pixel(x, bottom, onoff);
+        }
+        for (int y = 0; y < 5; y++) {
+            write_pixel(start, top + y, onoff);
+            write_pixel(stop,  top + y, onoff);
+        }
+    }
+}
+
+// Write active layer name
+static void render_layer_state(void) {
+  if (oled_horizontal) {
+    oled_set_cursor(0, 0);
+  } else {
+    oled_set_cursor(0, 15);
+  }
+  switch (get_highest_layer(layer_state)) {
+  case _MA:
+      oled_write(MA_name, false);
+      break;
+  case _L1:
+      oled_write(L1_name, false);
+      break;
+  case _L2:
+      oled_write(L2_name, false);
+      break;
+  case _L3:
+      oled_write(L3_name, false);
+      break;
```

Comment:
If you use a defines, then you could use `oled_write_P(PSTR(define-name))` instead of these.  And since it's an AVR board, that may make a difference, actually.